### ISSUE TITLE
RoundEnd event not called every round

### ISSUE THREAD

**Issue by micvbang** [markus-wa/demoinfocs-golang#54]
## Describe the bug
Hey Markus!

This is yet another I-found-a-bug-but-did-not-yet-investigate situation :slightly_smiling_face: 

I've found that in some demos the `RoundEnd` event is not triggered every round.

Since the internal score reported by the library is correct, at least `m_scoreTotal` is being called as we would expect.

## To Reproduce
Run the following code on this match: https://www.hltv.org/matches/2327343/mibr-vs-azio-ecs-season-6-north-america

```go
package main

import (
	"fmt"
	"log"
	"os"

	dem "github.com/markus-wa/demoinfocs-golang"
	"github.com/markus-wa/demoinfocs-golang/events"
	ex "github.com/markus-wa/demoinfocs-golang/examples"
)

func main() {
	f, err := os.Open(ex.DemoPathFromArgs())
	checkError(err)
	defer f.Close()

	p := dem.NewParser(f)

	_, err = p.ParseHeader()
	checkError(err)

	numRoundEnd := 1
	p.RegisterEventHandler(func(ev events.RoundEnd) {
		numRoundEnd++
	})

	numRoundEndOfficial := 1
	p.RegisterEventHandler(func(ev events.RoundEndOfficial) {
		numRoundEndOfficial++
	})

	err = p.ParseToEnd()
	checkError(err)

	fmt.Println("")
	fmt.Printf("Number of 'round end' events: %d\n", numRoundEnd)
	fmt.Printf("Number of 'round official end' events: %d\n", numRoundEndOfficial)
	fmt.Printf("Final score: %d:%d\n", p.GameState().TeamCounterTerrorists().Score(), p.GameState().TeamTerrorists().Score())
}

func checkError(err error) {
	if err != nil {
		log.Fatal(err)
	}
}
```

## Expected behavior
I would expect to see 26 `RoundEnd` events, as reported by HLTV

## Library version
I'm getting these results on `9916e2b4`

**Comment by markus-wa** [comment]
`numRoundEnd` & `numRoundEndOfficial` should start at zero, so there's also a `RoundEndOfficial` event missing.

Unfortunately the reason for this is that the demos don't contain these events (see below). So I don't think there's anything that can be done about it.

```
mwalt@DESKTOP-5DQ62Q0 MINGW64 ~/go/src/github.com/markus-wa/demoinfocs-golang (v1.0.0)
$ go test -run TestBadRounds -v -tags debugdemoinfocs | grep round_end | wc -l
19

mwalt@DESKTOP-5DQ62Q0 MINGW64 ~/go/src/github.com/markus-wa/demoinfocs-golang (v1.0.0)
$ go test -run TestBadRounds -v -tags debugdemoinfocs | grep round_officially_ended | wc -l
25
```

**Comment by micvbang** [comment]
Yeah, that's a shame!

I guess one solution that could potentially allow us to sort of circumvent this problem, would be to introduce a `SocreChanged` event, being called every time `m_scoreTotal` is updated. What do you think?

**Comment by markus-wa** [comment]
I think that's a great idea!

Feel free to submit a PR.

**Comment by markus-wa** [comment]
Marking as fixed for `v1.0.0`
